pkg/supervisor/supervisor.go

```diff
@@ -123,6 +127,14 @@ func (s *Supervisor) Supervise() error {
 		s.TimeoutRespawn = 5 * time.Second
 	}
 
+	_, err := os.Stat(s.PidFile)
+	if !os.IsNotExist(err) {
```

Comment:
This check is not necessary. We check the existence of the file when we open it.